sklearn/ensemble/_hist_gradient_boosting/_histogram_pool.py

```diff
@@ -0,0 +1,35 @@
+import numpy as np
+from .common import HISTOGRAM_DTYPE
+
+
+class HistogramsPool:
```

Comment:
I would suggest renaming to `HistogramPool` as it feels more correct. Although I am not a native English speaker so maybe I am wrong.

Reply:
Also it would be more consistent with the `histogram_pool` attribute of the grower and the module name.

Reply:
I now understand that each entry in the pool is actually a collection of several histograms (one per feature) and this was probably the motivation to use the plural in that class name. Still I find `HistogramPool` more natural.